src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -487,88 +491,119 @@ private int LaunchNode(string msbuildLocation, string commandLineArgs)
 
             CommunicationsUtilities.Trace("Launching node from {0}", msbuildLocation);
 
+            string exeName = msbuildLocation;
+
 #if RUNTIME_TYPE_NETCORE
             // Run the child process with the same host as the currently-running process.
-            string pathToHost;
-            using (Process currentProcess = Process.GetCurrentProcess()) pathToHost = currentProcess.MainModule.FileName;
+            exeName = GetCurrentHost();
             commandLineArgs = "\"" + msbuildLocation + "\" " + commandLineArgs;
 
-            ProcessStartInfo processStartInfo = new ProcessStartInfo();
-            processStartInfo.FileName = pathToHost;
-            processStartInfo.Arguments = commandLineArgs;
-            processStartInfo.CreateNoWindow = (creationFlags | BackendNativeMethods.CREATENOWINDOW) == BackendNativeMethods.CREATENOWINDOW;
-            processStartInfo.UseShellExecute = false;
-
-            Process process;
-            try
+            if (NativeMethodsShared.IsWindows)
             {
-                process = Process.Start(processStartInfo);
+                // Repeat the executable name _again_ because Core MSBuild expects it
+                commandLineArgs = exeName + " " + commandLineArgs;
```

Comment:
commandLineArgs is written to 3 times in this method. Maybe it would be more readable to extract its construction out